src/main/java/io/cryostat/net/web/http/api/v1/ReportGetHandler.java

```diff
@@ -119,10 +120,14 @@ public boolean isOrdered() {
     @Override
     public void handleAuthenticated(RoutingContext ctx) throws Exception {
         String recordingName = ctx.pathParam("recordingName");
+        List<String> queriedFilter = ctx.queryParam("filter");
+        String rawFilter = queriedFilter.isEmpty() ? null : queriedFilter.get(0);
```

Comment:
If `null` here was `""`, then the null-checks performed everywhere else within the actual handling logic wouldn't need to be done, right? It would be nice to handle that argument as always being non-nullable after this point.

Reply:
Even better, what if you used `RuleFilterParser` within each `Handler` implementation, and rather than passing the `String rawFilter` all the way deep into the call stack, you passed down something like an `Optional<Predicate<IRule>>`?

Reply:
I looked into this and it feels like it's somewhat messy to be passing Optional<Predicate<IRule>> because of things like the `SubprocessReportGenerator` needing String arguments to pass through main(String[] args), as well as needing to import those 3 dependencies in every file. To me it feels like it would be easier to parse the String at the end since a String is easy to send and receive and such, what are your thoughts?

Reply:
That's fair. At least doing the null-check at the point where you're receiving the client input seems like a good enough cleanup.

Reply:
Also, the [cryostat-reports] endpoint that I implemented expects a rawString like "rule1,rule2,..." and `RemoteReportGenerator` just sends a request to that endpoint with a filter form attribute so I am not sure about this. The part I use the `RuleFilterParser` is with the `SubprocessReportGenerator` where (I think) [cryostat-reports] doesn't exist so the functionality is emulated within that process, so then I have to directly pass in the `Predicate<IRule>` filter into the `generateReportInterruptibly` method.

Reply:
That's also a fair point. So, `String` it is.